doc/libraries/manual.rst

```diff
@@ -909,8 +921,77 @@ The following functions comprise the interface to the library:
   the ``FileReadyMsg`` sent to the ``opened`` callback after a file has been
   opened. Should only be called from a single PE, once per file.
 
+Parallel Input API
+~~~~~~~~~~~~~~~~~~
 
-Example
--------
+The following functions comprise the interface to the library for parallel file input:
+
+
+- Opening a file:
+
+  .. code-block:: c++
+
+     void Ck::IO::open(std::string path, CkCallback opened, Ck::IO::Options opts)
+
+  Open the given file with the options specified in ``opts``, and send a
+  ``FileReadyMsg`` (wraps a ``Ck::IO::File file``) to the ``opened`` callback
+  when the system is ready to accept session requests on that file. If the
+  specified file does not exist, it will be created. Should only be called from
+  a single PE, once per file.
+
+  ``Ck::IO::Options`` is a struct with the following input-relevant fields:
+
+  - ``numReaders`` - number of Buffer Chares, or aggregators. The user should chose this number to optimally decompose the read. Typically, chosing the number of Buffer Chares to be the number of PEs performs well.
+
+
+- Starting a read session:
+
+  Note there are two variants of the ``startReadSession`` function, a regular one and a variant which takes an additional argument allowing the user to map Buffer Chares to specified PEs in a round-robin fashion.
+
+  .. code-block:: c++
+
+    void startReadSession(File file, size_t bytes, size_t offset, CkCallback ready)
+
+  Prepare to read data from ``file``, in the window defined by ``size`` and
+  ``offset`` (both specified in bytes). On starting the session, the buffer 
+  chares begin eagerly reading all requested data into memory. The ready callback 
+  is invoked once these reads have been initiated (but they are not guaranteed to be complete at this point).
+
+  .. code-block:: c++
+
+    void startReadSession(File file, size_t bytes, size_t offset, CkCallback ready, std::vector<int> pes_to_map)
+
+  This function is similar to the previous one, but the extra argument pes_to_map allows the user to specify a list of PEs to map the Buffer Chares to. 
+  This argument should contain a sequence of numbers representing pes. The Buffer Chares will be mapped to the PEs in a round-robin fashion. 
+  This can be useful when the user has a specific decomposition in mind for the read.
+
+- Reading data:
+
+  .. code-block:: c++
+
+    void read(Session session, size_t bytes, size_t offset, char* data, CkCallback after_read);
+
+  This method is invoked to read data from the read session. If the data is not yet available, this request
+  will be buffered until the Buffer Chares can respond with the requested data. After the read finishes, the 
```

Comment:
I think there is a slight ambiguity here in that the user should rely on the callback regardless.  The function returns to the user when the request is submitted,  regardless of whether the data is already available, unlike a typical synchronous read call.

Reply:
I've updated this description. Let me know if it needs further clarification.

Reply:
looks good